packages/testlab/src/test-sandbox.ts

```diff
@@ -37,11 +38,16 @@ export class TestSandbox {
    * Will create a directory if it doesn't already exist. If it exists, you
    * still get an instance of the TestSandbox.
    *
-   * @param path - Path of the TestSandbox. If relative (it will be resolved relative to cwd()).
+   * @param rootPath - Path of the TestSandbox. If relative it will be resolved
+   * against the current directory.
+   * @param subdir - Subdirectory for the sandbox. If not provided, the sandbox
+   * will automatically generate a uuid as the name of the subdirectory.
    */
-  constructor(path: string) {
+  constructor(rootPath: string, subdir?: string) {
+    // Create a unique subdirectory so that sandboxes can be used in parallel
+    subdir = subdir ?? v1();
     // resolve ensures path is absolute / makes it absolute (relative to cwd())
-    this._path = resolve(path);
+    this._path = resolve(join(rootPath, subdir));
```

Comment:
Please note that `path.resolves` accepts an array of path segments, there is no need to call `path.join`. See https://nodejs.org/api/path.html#path_path_resolve_paths:

> The `path.resolve()` method resolves a sequence of paths or path segments into an absolute path.

```suggestion
    this._path = resolve(rootPath, subdir);
```